clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -9272,6 +9272,8 @@ def err_typecheck_expect_scalar_or_vector : Error<
   "a vector of such type is required">;
 def err_typecheck_expect_any_scalar_or_vector : Error<
   "invalid operand of type %0 where a scalar or vector is required">;
+def err_typecheck_expect_scalar_or_vector_not_type : Error<
```

Comment:
Do we need a new unique diagnostic for this or should we be using `err_typecheck_expect_any_scalar_or_vector` or `err_typecheck_expect_scalar_or_vector`?

In general we should try not to add unique diagnostics unless there really isn't an existing one that works.

In looking at this more closely I'm not sure we should have added `err_typecheck_expect_any_scalar_or_vector`, as we've basically added two _extremely similar_ diagnostics that probably should be combined.

Reply:
I think both those are confusing for the explicit purpose of rejecting `bool` or `vector<bool>`. The wording: `invalid operand of type 'bool' where a scalar or vector is required` or `invalid operand of type 'bool' where %1 or a vector of such type is required` implies `bool` or `bool4` is not a scalar or vector.

Reply:
Right, but we can combine them and use `%select` to cater to the specific message. Combining similar diagnostics helps keep Clang binary size from growing too much as the constant strings all need to get stored somewhere.